cosmovisor/cmd/cosmovisor/cmd/version.go

```diff
@@ -1,36 +1,85 @@
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
+	"strings"
 	"time"
 
+	"github.com/cosmos/cosmos-sdk/cosmovisor"
 	cverrors "github.com/cosmos/cosmos-sdk/cosmovisor/errors"
 	"github.com/rs/zerolog"
 )
 
-// Version represents Cosmovisor version value. Overwritten during build
-var Version = "1.1.0"
-
-// VersionArgs is the strings that indicate a cosmovisor version command.
-var VersionArgs = []string{"version", "--version"}
+var (
+	// FlagJSON formats the output in json
+	FlagJSON = "--json"
+	// Version represents Cosmovisor version value. Overwritten during build
+	Version = "1.1.0"
+	// VersionArgs is the strings that indicate a cosmovisor version command.
+	VersionArgs = []string{"version", "--version"}
+)
 
 // IsVersionCommand checks if the given args indicate that the version is being requested.
 func IsVersionCommand(arg string) bool {
 	return isOneOf(arg, VersionArgs)
 }
 
 // PrintVersion prints the cosmovisor version.
-func PrintVersion() error {
+func PrintVersion(args []string) error {
+	for _, arg := range args {
+		if strings.Contains(arg, FlagJSON) {
+			return printVersionJSON(args)
+		}
+	}
+
+	return printVersion(args)
+}
+
+func printVersion(args []string) error {
 	fmt.Println("Cosmovisor Version: ", Version)
 
-	if err := Run([]string{"version"}); err != nil {
-		// Check the config and output details or any errors.
-		// Not using the cosmovisor.Logger in order to ignore any level it might have set,
-		// and also to not have any of the extra parameters in the output.
-		output := zerolog.ConsoleWriter{Out: os.Stdout, TimeFormat: time.Kitchen}
-		logger := zerolog.New(output).With().Timestamp().Logger()
-		cverrors.LogErrors(logger, "Can't run APP version", err)
+	if err := Run(append([]string{"version"}, args...)); err != nil {
```

Comment:
what if arg already had `version`?

Reply:
I thought of deduplicating them, but I realized that the application does not read arguments:

- https://github.com/cosmos/cosmos-sdk/blob/main/version/command.go#L19 

As cosmovisor depends on the behavior of the commands from the sdk, do you think it is still necessary to deduplicate it there?